protocol/src/types/block.rs

```diff
@@ -19,8 +19,34 @@ pub const MAX_FEE_HISTORY: u64 = 1024;
 pub const MAX_RPC_GAS_CAP: u64 = 50_000_000;
 pub const BASE_FEE_PER_GAS: u64 = 0x539;
 
+#[derive(Serialize, Deserialize, Default, Copy, Clone, Debug, PartialEq, Eq)]
+pub enum BlockVersion {
+    #[default]
+    V0,
+}
+
+impl From<BlockVersion> for u8 {
+    fn from(value: BlockVersion) -> Self {
+        match value {
+            BlockVersion::V0 => 0,
+        }
+    }
+}
```

Comment:
Pseudocode, handwritten, untested.

```suggestion
#[derive(Serialize, Deserialize, Default, Copy, Clone, Debug, PartialEq, Eq)]
#[repr(u8)]
pub enum BlockVersion {
    #[default]
    V0 = 0,
}
```

Reply:
There's also serde_repr if we want to serialize the version as number.

Reply:
I do not think serialize as `uint` is suitable. The `BlockVersion` imply the `Into<u8>` trait for rlp encodable and decodable. When serialized as json format. it should be like this:

```json
{
	"version": "V0",
}
```

Reply:
> When serialized as json format. it should be like this:
> 
> ```json
> {
> 	"version": "V0",
> }
> ```

In fact, it's a bad format.

I agree that the version should be a string, due to the poor types system of Javascript.
But it should not be `V`-prefixed.

eg:

- His first name is Wang

  We won't say: his first name is "1st name Wang".

> No, “v1.2.3” is not a semantic version. However, prefixing a semantic version with a “v” is a common way (in English) to indicate it is a version number. Abbreviating “version” as “v” is often seen with version control.

Ref: [Is “v1.2.3” a semantic version?](https://semver.org/#is-v123-a-semantic-version)

Reply:
This `V` is not a prefix, it is a part of the version name. It can be change as `ALPHA`, `BETA` and so on. Besides it is not a semantic version.